src/test/java/de/tum/in/www1/artemis/util/QueryCountAssert.java

```diff
@@ -0,0 +1,58 @@
+package de.tum.in.www1.artemis.util;
+
+import org.assertj.core.api.AbstractAssert;
+
+public class QueryCountAssert<T, E extends Exception> extends AbstractAssert<QueryCountAssert<T, E>, ThrowingProducer<T, E>> {
+
+    private final HibernateQueryInterceptor interceptor;
+
+    protected QueryCountAssert(HibernateQueryInterceptor interceptor, ThrowingProducer<T, E> tProducer, Class<?> selfType) {
+        super(tProducer, selfType);
+        this.interceptor = interceptor;
+    }
+
+    public static <T, E extends Exception> QueryCountAssert<T, E> assertThatDb(HibernateQueryInterceptor interceptor, ThrowingProducer<T, E> call) {
+        return new QueryCountAssert<>(interceptor, call, QueryCountAssert.class);
+    }
+
+    /**
+     * Asserts that the amount of database queries realized during a REST call exactly match the expected amount
```

Comment:
```suggestion
     * Asserts that the number of database queries executed during a REST call exactly matches the expected count
```
The number of queries is countable.